### PR TITLE
[wip]Anomaly Core Production

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

https://hackmd.io/@tgstation/r1tzxpwPL

Basically that.
Raw anomaly cores
Cargo crates for them (currently 5k credits each, please tell me if this is good I balanced it relative to other crates I have no clue what the current cargo "meta"/power levels are)

- [x] Finish the explosive/implosion compressor attackby and implosion procs 

- [ ] Map in the implosion compressor

- [ ] Map in the roundstart 2 cores each type for science

Current values are set to maxcap at the 8th core, and 4 light radius (1/2/4 explosion required) at the first core. I think this is a bit low but I'm just going off the document.

**The implosion compressor consumes the TTV + tanks on an attempted detonation, regardless of success.** However, the "bare minimum" of checks are included (it won't let you insert incomplete TTVs. It will however, let you insert TTVs that have no explosive potential (test your bombs like you test your code ;) ), and will not waste the raw core if you fail to make a high enough radius.

Props to @ArcaneMusic for the machine sprite! 

This currently doesn't do something like not letting cargo order it if there's already the max amount made, so 

- [ ] Disable the anomaly core crates when the max amount is already made so cargonians aren't going to get baited into making more.

## Why It's Good For The Game
image

Start of the anomaly core system I guess, gives toxins a use other than doing the quirky 50k maxcap tritium speedrun rush (or was that removed already?).

Yadda yadda allows locking items behind it and another method of getting these other than supermatter overloads (I don't know if that is already gone either by the by).

## Changelog
:cl:
add: An explosion compressor has been added to science. It makes raw anomaly cores into proper anomaly cores using prepared TTVs of a specific required blast radius.
/:cl:

### CODE REVIEW

File: code/modules/assembly/signaler.dm [hunk 1]
```diff
@@ -178,70 +178,6 @@
 		return
 	return ..(signal)
 
-// Embedded signaller used in anomalies.
```
Comment (kevinz000) [inline code/modules/assembly/signaler.dm:181]: moved anomaly stuff to its own file under sci btw

**Comment by optimumtact** [comment]
how are you so fast

**Comment by Iatots** [comment]
He is not held back by bigoted code anymore

**Comment by TheVekter** [comment]
NUT. Can you discord me when this gets merged? Can't touch my stuff much until you're done monkeying around with it or I'll conflict all over the place.

**Comment by Arturlang** [comment]
Cant you just reliably produce SM anomalies using the SM?

**Comment by TheVekter** [comment]
I think we've kind of agreed that putting the SM anywhere near delaminating, even _if_ it's stable, isn't how we want to go about this.

Also I'm not 100% sure but anomalies might not drop cores anymore after this.

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -1408,6 +1408,51 @@
 	crate_name = "plasma assembly crate"
 	crate_type = /obj/structure/closet/crate/secure/plasma
 
+/datum/supply_pack/science/raw_flux_anomaly
+	name = "Raw Flux Anomaly"
+	desc = "The raw core of a flux anomaly, ready to be implosion-compressed into a powerful artifact."
```
Comment (Fikou) [inline code/modules/cargo/packs.dm:1413]: description of these should probably mention them being access locked to toxins storage

**Comment by Tlaltecuhtli** [comment]
> I think we've kind of agreed that putting the SM anywhere near delaminating, even _if_ it's stable, isn't how we want to go about this. IMO it's because it makes it too easy for someone to delam it and say "i WAS TrYINg tO GeNerAte AnOmAliES".
> 
> Also I'm not 100% sure but anomalies might not drop cores anymore after this.

dangerous setups so the sm had some content was the whole point when kor added craftable reactive armors and has been fine until now? where are these pebble brains using it as excuse to griff? just notice how they put a random can in there and jobban anyway for doing things without knowing?
they way to farm anomalies are easily visible by the fact there are 102390213 heatsink pipes in the cooling loop, which makes sm go to -247 with any gas you put there, if someone is putting a co2 can, turning off vents its visible from a mile he is there to griff

File: code/modules/research/anomaly/raw_anomaly.dm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		if(ANOMALY_TYPE_BLUESPACE)
+			path = /obj/item/assembly/signaler/anomaly/bluespace
+		if(ANOMALY_TYPE_PYRO)
+			path = /obj/item/assembly/signaler/anomaly/pyro
+		if(ANOMALY_TYPE_FLUX)
+			path = /obj/item/assembly/signaler/anomaly/flux
+		if(ANOMALY_TYPE_GRAVITATIONAL)
+			path = /obj/item/assembly/signaler/anomaly/grav
+		if(ANOMALY_TYPE_VORTEX)
+			path = /obj/item/assembly/signaler/anomaly/vortex
```
Comment (bobbahbrown) [inline code/modules/research/anomaly/raw_anomaly.dm:68]: why not store these typepaths in the raw_anomaly_core type

**Comment by zxaber** [comment]
We're moving away from SM anomalies because oranges wants there to be a hard-limited number of cores per round. This also gives Science back some content what with R&D being yeeted to the bridge.

File: code/modules/research/anomaly/raw_anomaly.dm (hunk 3 above)
Comment (kevinz000) [reply code/modules/research/anomaly/raw_anomaly.dm:68]: tbh yeah ill throw the typepaths into the defines , there's not really any reason to not use 'em rn

**Comment by optimumtact** [comment]
@Tlaltecuhtli please read the attached hackmd document before commenting

**Comment by Spepbuthewranglesgithub** [comment]
dear god, someone might actually use the portal gun now.


_if they don't use all the anomaly cores on actually useful equipment, anyway._

**Comment by TheVekter** [comment]
That's the idea!

**Comment by kevinz000** [comment]
"yet another kevinz wip pr"

but no i didn't forget or anything, waiting on the explosion inhibitor pr

**Comment by Bawhoppen** [comment]
How about we rename them from being anomaly cores? It doesn't really make sense to call them that anymore, if they're such a routine occurrence now (while they were super common before, them appearing every 5 minutes was just a game mechanic, in the same way that brig timers are in minutes, rather than days and weeks). There's alot of potential for cool names we could replace them with which would greatly add to the flavor.

**Comment by kevinz000** [comment]
@Bawhoppen you do that i'm the most uncreative brick here

**Comment by TheVekter** [comment]
Anti-stale comment

**Comment by kevinz000** [comment]
right I just need to map it in

**Comment by Bawhoppen** [comment]
I couldn't come up with a neat scifi name, but I'd think you may wanna just call them Physics Cores if nothing else.

**Comment by SpaceManiac** [comment]
it's stale

**Pull request closed without merging**